dll/win32/framedyn/WbemTime.cpp

```diff
@@ -0,0 +1,522 @@
+    }
+    else
+    {
+        m_uTime = INVALID_TIME;
+    }
+
+    return *this;
+}
+
+WBEMTimeSpan WBEMTime::operator- (const WBEMTime &sub)
+{
+    WBEMTimeSpan rt;
+
+    if (IsOk() && sub.IsOk() && m_uTime >= sub.m_uTime)
+    {
+        rt.m_Time = m_uTime - sub.m_uTime;
+    }
+
+    return rt;
+}
+
+WBEMTime WBEMTime::operator- (const WBEMTimeSpan &sub) const
+{
+    WBEMTime rt;
+
+    if (IsOk() && sub.IsOk() && m_uTime >= sub.m_Time)
+    {
+        rt.m_uTime = m_uTime - sub.m_Time;
+    }
+
+    return rt;
+}
+
+const WBEMTime& WBEMTime::operator-=(const WBEMTimeSpan &sub)
+{
+    if (IsOk() && sub.IsOk() && m_uTime >= sub.m_Time)
+    {
+        m_uTime -= sub.m_Time;
+    }
+    else
+    {
+        m_uTime = INVALID_TIME;
+    }
+
+    return *this;
+}
+
+// assignment operator overload method takes a CIM date time format parameter.
+const WBEMTime& WBEMTime::operator= (const BSTR bstrDMTFFormat)
+{
+    m_uTime = INVALID_TIME;
+
+    if (bstrDMTFFormat != NULL && wcslen(bstrDMTFFormat) == 25 && bstrDMTFFormat[14] == L'.')
+    {
+        if (bstrDMTFFormat[21] == L'+' || bstrDMTFFormat[21] == L'-')
+        {
+            SetDMTF(bstrDMTFFormat);
+        }
+    }
+
+    return *this;
+}
+
+// assignment operators have been overloaded to facilitate conversions between various Windows and ANSI C run-time time formats.
+const WBEMTime& WBEMTime::operator= (const SYSTEMTIME &st)
+{
+    FILETIME ft;
+
+    if (SystemTimeToFileTime(&st, &ft))
+    {
+        *this = ft;
+    }
+    else
+    {
+        m_uTime = INVALID_TIME;
+    }
+
+    return *this;
+}
+
+// assignment operator overload method takes a FILETIME parameter. The WBEMTime class assignment operators have been overloaded to facilitate conversions between various Windows and ANSI C run-time time formats.
+const WBEMTime& WBEMTime::operator= (const FILETIME &ft)
+{
+    m_uTime = *(ULONGLONG*)&ft;
+    return *this;
+}
+
+// assignment operators have been overloaded to facilitate conversions between various Windows and ANSI C run-time time formats.
+const WBEMTime& WBEMTime::operator= (const struct tm &tmin)
+{
+    SYSTEMTIME sysTime;
+
+    tm_to_SYSTEMTIME(&tmin, &sysTime);
+
+    *this = sysTime;
+
+    return *this;
+}
+
+// assignment operators have been overloaded to facilitate conversions between various Windows and ANSI C run-time time formats.
+const WBEMTime& WBEMTime::operator= (const time_t &t)
+{
+    FILETIME ft;
+    time_t_to_FILETIME(t, &ft);
+
+    *this = ft;
+    return *this;
+}
+
+// gets the time as a BSTR value in CIM Date and Time Format.
+BSTR WBEMTime::GetBSTR(void) const throw (CHeap_Exception)
+{
+    return GetDMTF(false);
+}
+
+// gets the time as an ANSI C run-time struct tm structure.
+BOOL WBEMTime::GetStructtm(struct tm *ptm) const
+{
+    if (ptm == NULL)
+    {
+        return false;
+    }
+
+    SYSTEMTIME sysTime;
+    if (!GetSYSTEMTIME(&sysTime) || sysTime.wYear < 1900)
+    {
+        return false;
+    }
+
```

Comment:
As for `WBEMTime::GetStructtm` and any other function declared as booleans using Windows data type `BOOL` you must return their condition values accordingly as TRUE/FALSE not true/false or else (as Hermes Belusca explained above).

Reply:
I've fixed.